src/rules/semicolonRule.ts

```diff
@@ -222,8 +222,19 @@ class SemicolonWalker extends Lint.AbstractWalker<Options> {
                 case ts.SyntaxKind.RegularExpressionLiteral:
                     break;
                 default:
-                    this.reportUnnecessary(node.end - 1);
+                    if (!this.isFollowedByStatement(node)) {
+                        this.reportUnnecessary(node.end - 1);
+                    }
             }
         }
     }
+
+    private isFollowedByStatement(node: ts.Statement): boolean {
+        const nextStatement = utils.getNextStatement(node);
+        if (nextStatement === undefined) {
+            return false;
+        }
+        return ts.getLineAndCharacterOfPosition(this.sourceFile, node.end).line
```

Comment:
This should be a utility somewhere. `curlyRule.ts` also has an `areOnSameLine` helper. And there was some other situation I wanted a helper for this that I can't recall right now.
I think the best way to do this would be to look for a "\n" character between the end of node1 and the start of node2.
The current method would be tricked by a multi-line statement followed by another, i.e.:
```ts
const x = f(() =>
    0); const y = 2;
```